pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -272,17 +280,25 @@ func (c *Cluster) startProvisioningOverPVCs(config *provisionConfig) {
 		//
 		// The logic will get a bit more complex if we plan support for a third device for "block.wal"
```

Comment:
I don't see how we can rely on the ordering of the templates. `i` for the data, `i+1` for metadata, and `i+2` for wal wouldn't be reliable. The reason that we name the templates is so that we don't have to rely on the order. 
@leseb Why was the ordering still there since we named the templates?